matrix-synapse-riot/terraform/matrix-server/bootstrap-post-dns.sh

```diff
@@ -0,0 +1,103 @@
+sed -i -e "s/CHATDNS/$DOMAIN_NAME/g" /etc/nginx/sites-available/matrix.$DOMAIN_NAME
+ln -s /etc/nginx/sites-available/matrix.$DOMAIN_NAME /etc/nginx/sites-enabled/matrix.$DOMAIN_NAME
+ln -s /etc/nginx/sites-available/chat.$DOMAIN_NAME /etc/nginx/sites-enabled/chat.$DOMAIN_NAME
+
+# Install Dehydrated 
+cd /opt/
+wget https://github.com/lukas2511/dehydrated/releases/download/v$DEHYDRATED_VERSION/dehydrated-$DEHYDRATED_VERSION.tar.gz -O /opt/dehydrated-$DEHYDRATED_VERSION.tar.gz
+tar xf dehydrated-$DEHYDRATED_VERSION.tar.gz
+rm dehydrated-$DEHYDRATED_VERSION.tar.gz
+chmod 700 /opt/dehydrated-$DEHYDRATED_VERSION
+cd /opt/dehydrated-$DEHYDRATED_VERSION
+cp /tmp/matrix-server/dehydrated-config /opt/dehydrated-$DEHYDRATED_VERSION/config
+cp /tmp/matrix-server/dehydrated-hooks /opt/dehydrated-$DEHYDRATED_VERSION/hooks.sh
+echo -n "chat.$DOMAIN_NAME matrix.$DOMAIN_NAME" > /opt/dehydrated-$DEHYDRATED_VERSION/domains.txt
+chmod +x /tmp/matrix-server/dehydrated-hooks /opt/dehydrated-$DEHYDRATED_VERSION/hooks.sh
+ln -s /opt/dehydrated-$DEHYDRATED_VERSION /opt/dehydrated
+cd /opt/dehydrated
+sed -i -e "s/TOKENSTRING/$DO_TOKEN/g" hooks.sh
+sed -i -e "s/DODOMAIN/$DOMAIN_NAME/g" hooks.sh
+
+# Register to Let's Encrypt
+./dehydrated --register --accept-terms
+
+cp /tmp/matrix-server/dehydrated-renew.sh /usr/local/sbin/dehydrated-renew.sh
+
+# Configure auto-renewals
+sh -c '(crontab -l 2>/dev/null; echo "@weekly /usr/local/sbin/dehydrated-renew.sh") | crontab -'
+
+############
+# iptables #
+############
+cp /tmp/matrix-server/ip6tables-rules /etc/iptables/rules.v6
+cp /tmp/matrix-server/iptables-rules /etc/iptables/rules.v4
+
+#######################
+# Make non-root users #
+#######################
+useradd -s /bin/bash -N -g users -G sudo -m sysadmin
+useradd -s /bin/bash -N -g users -m synapse
+passwd=$(dd if=/dev/urandom bs=1M count=500 | sha256sum | awk '{ print $1 }')
+echo "sysadmin:$passwd" | chpasswd
+echo -n $passwd > /tmp/matrix-server/passwd-sysadmin
+mkdir -m 700 /home/sysadmin/.ssh/
+cp /root/.ssh/authorized_keys /home/sysadmin/.ssh/authorized_keys
+chown -R sysadmin:users /home/sysadmin/.ssh/
+chage -d 0 sysadmin
+
+########
+# PSQL #
+########
+# Generate an password for synapse database.
+DATAPASS=$(dd if=/dev/urandom bs=1M count=500 | sha256sum | awk '{ print $1 }')
+su -c "/tmp/matrix-server/psql-setup.sh $DATAPASS" postgres
+
+###########
+# Synapse #
+###########
+su -c "/tmp/matrix-server/synapse-setup.sh $DOMAIN_NAME $DATAPASS" synapse
+
+########
+# Riot #
+########
+cd /var/www/
+wget https://github.com/vector-im/riot-web/releases/download/v$RIOT_VERSION/riot-v$RIOT_VERSION.tar.gz
+mkdir -p /var/www/chat.$DOMAIN_NAME/public
+tar xf riot-v$RIOT_VERSION.tar.gz -C /var/www/chat.$DOMAIN_NAME/public --strip-components 1
+cp /tmp/matrix-server/riot-config.json /var/www/chat.$DOMAIN_NAME/public/config.json
+sed -i -e "s/DOMAIN/$DOMAIN_NAME/g" /var/www/chat.$DOMAIN_NAME/public/config.json
```

Comment:
I prefer using two underscores around macros to avoid unintended replacements `__DOMAIN_NAME__` and try to match them exactly as the var name that will eventually replace them to make code more readable